allennlp/training/model.py

```diff
@@ -0,0 +1,94 @@
+from typing import Any, List, Dict, Optional, Union
+import torch
+
+
+class Model(torch.nn.Module):
```

Comment:
Thanks for putting this together. Is the intention that a user will provide on of these `Model` implementations, specify their data with the data API, then the Trainer (or other allennlp components) will handle the training loop?